### PR TITLE
Common properties, add security to admin-client (using `configure` sub-command)

### PR DESCRIPTION
**Pull request by im-konge**
This PR adds security configuration into the admin-client CLI tool, using the `configure` sub-command, which is also added by this PR.

The PR does following:

- moves `KafkaClientsConfiguration`, `SaslType`, and other classes from the `clients` module to new module - `clients-common`, from where it can be used by admin-client and also clients
- splits `KafkaProperties` to two classes -> `BasicKafkaProperties` configuring basic and common Kafka properties for the clients, and `KafkaProperties` (name is kept) for the Kafka properties for producer, consumer, and streams (which are not required by admin client)
- adds classes for handling the `configure` sub-command (with `oauth`, `ssl`, `sasl`, `common` sub-commands)
- moves the tests related to `KafkaClientsConfiguration` and `(Basic)KafkaProperties` to the `clients-common` module
- adds method for loading config file, env variables, and overall handling of the configuration
- updates `admin/README.md`

### CODE REVIEW

File: admin/src/main/java/io/strimzi/arguments/configure/OauthCommand.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @CommandLine.Option(names = {"--client-secret", "-cs"}, description = "Secret of the client used in Oauth")
+    String oauthClientSecret;
+
+    @CommandLine.Option(names = {"--access-token", "-at"}, description = "Access token for accessing Oauth")
+    String oauthAccessToken;
+
+    @CommandLine.Option(names = {"--refresh-token", "-rt"}, description = "Oauth refresh token")
+    String oauthRefreshToken;
+
+    @CommandLine.Option(names = {"--token-endpoint", "-teu"}, description = "Oauth token endpoint URI for obtaining new token")
```
Comment (jankalinic) [inline admin/src/main/java/io/strimzi/arguments/configure/OauthCommand.java:31]: what about `-te` as the` --token-endpoint` option also does not contain` -uri`

File: admin/src/main/java/io/strimzi/arguments/configure/SaslCommand.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import io.strimzi.SaslType;
+import io.strimzi.arguments.CommandInterface;
+import io.strimzi.configuration.ConfigurationConstants;
+import io.strimzi.utils.ConfigurationUtils;
+import picocli.CommandLine;
+
+import java.util.Properties;
+
+/**
+ * Sub-command for configuring SASL related configuration for admin-client
```
Comment (jankalinic) [inline admin/src/main/java/io/strimzi/arguments/configure/SaslCommand.java:16]:
```suggestion
 * Subcommand for setting up SASL related configuration of admin-client
```

File: admin/src/main/java/io/strimzi/arguments/configure/SaslCommand.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    static class Configuration {
+        @CommandLine.Option(names = "--jaas-config", description = "SASL JAAS config that should be passed to the configuration", required = true)
+        String saslJaasConfig;
+
+        @CommandLine.ArgGroup(exclusive = false, multiplicity = "1")
+        User user;
+    }
+
+    static class User {
+        @CommandLine.Option(names = "--user", description = "Username of the user", required = true)
```
Comment (jankalinic) [inline admin/src/main/java/io/strimzi/arguments/configure/SaslCommand.java:37]:
```suggestion
        @CommandLine.Option(names = "--user", description = "Name of the user", required = true)
```

Comment (jankalinic) [inline admin/src/main/java/io/strimzi/arguments/configure/SaslCommand.java:37]: how about `--username` option instead

File: admin/src/main/java/io/strimzi/arguments/configure/SslCommand.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import io.strimzi.arguments.CommandInterface;
+import io.strimzi.configuration.ConfigurationConstants;
+import io.strimzi.constants.Constants;
+import io.strimzi.utils.ConfigurationUtils;
+import picocli.CommandLine;
+
+import java.util.Properties;
+
+/**
+ * Sub-command for configuring SSL related configuration for admin-client
```
Comment (jankalinic) [inline admin/src/main/java/io/strimzi/arguments/configure/SslCommand.java:16]:
```suggestion
 * Subcommand for setting up SSL related configuration of admin-client
```

**Review by jankalinic — APPROVED** [review]
just small nits

**Review by Frawless — APPROVED** [review]

**Pull request merged**